File: openmp/testsuite/c/omp_flush.c
<ompts:test>
<ompts:testdescription>Test which checks the omp flush directive.</ompts:testdescription>
<ompts:ompversion>2.0</ompts:ompversion>
<ompts:directive>omp flush</ompts:directive>
<ompts:dependences>omp barrier</ompts:dependences>
<ompts:testcode>
#include <stdio.h>
#include <unistd.h>

#include "omp_testsuite.h"
#include "omp_my_sleep.h"

int <ompts:testcode:functionname>omp_flush</ompts:testcode:functionname> (FILE * logFile)
{
    <ompts:orphan:vars>
	int result1;
	int result2;
	int dummy;
    </ompts:orphan:vars>

	result1 = 0;
	result2 = 0;

#pragma omp parallel
    {
	int rank;
	rank = omp_get_thread_num ();

#pragma omp barrier
	if (rank == 1) {
	    result2 = 3;
	    <ompts:orphan>
		<ompts:check>#pragma omp flush (result2)</ompts:check>
		dummy = result2;
	    </ompts:orphan>
	}

	if (rank == 0) {
	    <ompts:check>my_sleep(SLEEPTIME_LONG);</ompts:check>
	    <ompts:orphan>
		<ompts:check>#pragma omp flush (result2)</ompts:check>
		result1 = result2;
	    </ompts:orphan>
	}
    }	/* end of parallel */

    return ((result1 == result2) && (result2 == dummy) && (result2 == 3));
}
</ompts:testcode>
</ompts:test>

File: openmp/testsuite/fortran/omp_flush.f
<ompts:test>
<ompts:testdescription>Test which checks the omp flush directive.</ompts:testdescription>
<ompts:ompversion>2.0</ompts:ompversion>
<ompts:directive>omp flush</ompts:directive>
<ompts:dependences>omp barrier</ompts:dependences>
<ompts:testcode>
        INTEGER FUNCTION <ompts:testcode:functionname>omp_flush</ompts:testcode:functionname>()
        IMPLICIT NONE
        INTEGER result1, dummy, rank
        INTEGER omp_get_thread_num
        <ompts:orphan:vars>
        INTEGER result2
        COMMON /orphvars/ result2
        </ompts:orphan:vars>
        result1=0
        result2=0
!$omp parallel private(rank)
        rank = omp_get_thread_num()
!$omp barrier
        IF ( rank .EQ. 1 ) THEN
          result2 = 3
          <ompts:orphan>
          <ompts:check>
!$omp flush(result2)
          </ompts:check>
          </ompts:orphan>
          dummy = result2
        END IF
        IF ( rank .EQ. 0 ) THEN
          call sleep(1)
          <ompts:orphan>
          <ompts:check>
!$omp flush(result2)
          </ompts:check>
          </ompts:orphan>
          result1 = result2
        END IF
!$omp end parallel

!        PRINT *,"1:", result1, "2:", result2, "dummy", dummy
        IF ( (result1 .EQ. result2) .AND. (result2 .EQ. dummy) .AND.
     &       (result2 .EQ. 3) ) THEN
           <testfunctionname></testfunctionname> = 1
        ELSE
           <testfunctionname></testfunctionname> = 0
        END IF
      END
</ompts:testcode>
</ompts:test>
